src/librustc/diagnostics.rs

```diff
@@ -1886,7 +1886,51 @@ This explicitly states that you expect the trait object `SomeTrait` to
 contain references (with a maximum lifetime of `'a`).
 
 [1]: https://github.com/rust-lang/rfcs/pull/1156
-"##
+"##,
+
+E0454: r##"
+A link name was given with an empty name. Erroneous code example:
+
+```
+#[link(name = "")] extern {} // error: #[link(name = "")] given with empty name
+```
+
+The rust compiler cannot link to an external library if you don't give it its
+name. Example:
+
+```
+#[link(name = "some_lib")] extern {} // ok!
+```
+"##,
+
+E0458: r##"
+An unknown kind type was specified. Erroneous code example:
+
+```
+#[link(kind = "unknown")] extern {} // error: unknown kind: `unwknown`
```

Comment:
Let's use a name like "unicorn" here, when we use "unknown" as a literal it can get confusing.

Also reminds me of "undefined" in JS.